msal/application.py

```diff
@@ -498,7 +512,7 @@ def _get_app_metadata(self, environment):
 
     def _acquire_token_silent_by_finding_specific_refresh_token(
             self, authority, scopes, query,
-            rt_remover=None, break_condition=lambda response: False, **kwargs):
+            rt_remover=None, break_condition=lambda response: False, force_refresh=False, **kwargs):
```

Comment:
You can use the **kwargs to send any additional parameters to this function instead of adding a new parameter

Reply:
Yes Abhi and I briefly talked on this one yesterday. Either the implicit `**kwargs` or the explicit `force_refresh` here would work. Now on closer inspection, I think it is OK to use an explicit `force_refresh` here, because the later inside this function, we do use that parameter. But then, some extra changes are needed:

1. You would need to pull in the latest dev branch which contains a bugfix relevant to this feature.
2. You would need to modify an "outer" level method `_acquire_token_silent_from_cache_and_possibly_refresh_it()` to wire up this `force_refresh`, only then it would be available here inside the innermost method.
3. Given that these intrinsic complexity of the acquire_token_silent(...) family, I would suggest let's see if we can add a test case for this function. You can find the [pattern to test in this pre-existing test case](https://github.com/AzureAD/microsoft-authentication-library-for-python/blob/0.7.0/tests/test_application.py#L75-L81). That would have helped you to catch the above issue.
4. Lastly, per PEP 8 (even though we have not strictly enforce it, yet), I would suggest you move this new parameter into its own line, so that the current line remains within the recommend [max line length](https://www.python.org/dev/peps/pep-0008/#maximum-line-length).